src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl

```diff
@@ -8,151 +8,134 @@ if [ ! -d "$PIP_CACHE_DIR" ];then
 fi
 
 {{ if VirtualEnvironmentName | IsNotBlank }}
+	{{ if PackagesDirectory | IsNotBlank }}
+		if [ -d "{{ PackagesDirectory }}" ]
+		then
+			rm -fr "{{ PackagesDirectory }}"
+		fi
+	{{ end }}
 
-{{ if PackagesDirectory | IsNotBlank }}
-if [ -d "{{ PackagesDirectory }}" ]
-then
-	rm -fr "{{ PackagesDirectory }}"
-fi
-{{ end }}
-
-VIRTUALENVIRONMENTNAME={{ VirtualEnvironmentName }}
-VIRTUALENVIRONMENTMODULE={{ VirtualEnvironmentModule }}
-VIRTUALENVIRONMENTOPTIONS={{ VirtualEnvironmentParameters }}
-zippedVirtualEnvFileName={{ CompressedVirtualEnvFileName }}
+	VIRTUALENVIRONMENTNAME={{ VirtualEnvironmentName }}
+	VIRTUALENVIRONMENTMODULE={{ VirtualEnvironmentModule }}
+	VIRTUALENVIRONMENTOPTIONS={{ VirtualEnvironmentParameters }}
+	zippedVirtualEnvFileName={{ CompressedVirtualEnvFileName }}
 
-echo "Python Virtual Environment: $VIRTUALENVIRONMENTNAME"
+	echo "Python Virtual Environment: $VIRTUALENVIRONMENTNAME"
 
-echo Creating virtual environment ...
-$python -m $VIRTUALENVIRONMENTMODULE $VIRTUALENVIRONMENTNAME $VIRTUALENVIRONMENTOPTIONS
+	echo Creating virtual environment...
+	$python -m $VIRTUALENVIRONMENTMODULE $VIRTUALENVIRONMENTNAME $VIRTUALENVIRONMENTOPTIONS
 
-echo Activating virtual environment ...
-source $VIRTUALENVIRONMENTNAME/bin/activate
+	echo Activating virtual environment...
+	source $VIRTUALENVIRONMENTNAME/bin/activate
 
-if [ -e "requirements.txt" ]
-then
-	echo
```

Comment:
Since we are using the `-m` option, we no longer need to run these steps of upgrade. We will be using the pip version which was built as part of Python itself.